### PR TITLE
Fix master branch

### PR DESCRIPTION
**Pull request by chansbtran**
### Summary
Merging the buffer PR into Trihydro's fork of GH does not build successfully. There have been a couple issues since the BufferResource update.

1. FlagEncoder has been removed. [Reference link to GH PR](https://github.com/graphhopper/graphhopper/pull/2611)
2. ~Trihydro fork of GH is out-of-date~
3. Running the buffer endpoint results in not able to find route

### Solution
- Pull from GH main to be more up-to-date
- Update BufferResource to remove FlagEncoder
- BufferResource now includes both StreetName and StreetRef. This is because the user may input a route name that is exclusive to one of the lists. By including both, we allow a wider range of available route names.
- Uncomment unit test and ensure it runs properly
- Code cleanup through abstracting duplicate functionality and resolve Java warnings

### Notes
The solution to fixing the BufferResource is actually fairly simple, just to concatenate the list of street names and refs. The list of route names of those lists seem to be exclusive so by combining the fetch, we can expand the routes that the user is actually trying to query by.

### TODOs 
- [x] ~Remove test API endpoint~
- [x] ~Test to make sure buffer endpoint is working properly~
- [x] ~Add/update unit tests~

### CODE REVIEW

File: web-bundle/src/main/java/com/graphhopper/resources/BufferResource.java [hunk 1]
```diff
@@ -260,31 +242,51 @@ private LineString computeBufferSegment(BufferFeature startFeature, String roadN
...
+     */
+    private List<String> sanitizeRoadNames(String roadNames) {
+        // Return empty list if roadNames is null
+        if (roadNames == null) {
+            return new ArrayList<>();
         }
 
+        List<String> separatedNames = Arrays.asList(roadNames.split(","));
+        separatedNames.replaceAll(s -> {
+            // TODO: do we want to remove all spaces from the names? There could be a mismatch between "I 80" "I-80" "I80" etc.
```
Comment (payneBrandon) [inline web-bundle/src/main/java/com/graphhopper/resources/BufferResource.java:273]: Probably appropriate in a method that sanitizes the names to return them in a consistent format. We'll want to update the description accordingly

**Review by payneBrandon — CHANGES_REQUESTED** [review]
Just the one comment

**Review by tsweckard — APPROVED** [review]

File: web-bundle/src/main/java/com/graphhopper/resources/BufferResource.java (hunk 1 above)
Comment (chansbtran) [reply web-bundle/src/main/java/com/graphhopper/resources/BufferResource.java]: Updated!

**Review by payneBrandon — APPROVED** [review]
lgtm!

**Pull request merged**